assets/js/googlesitekit/data/create-settings-store.test.js

```diff
@@ -0,0 +1,453 @@
+					),
+				);
+
+				const settings = select.getSettings();
+
+				expect( fetch ).toHaveBeenCalledTimes( 1 );
+				expect( settings ).toEqual( response );
+
+				expect( fetch ).toHaveBeenCalledTimes( 1 );
+				expect( select.getSettings() ).toEqual( settings );
+			} );
+
+			it( 'returns client settings even if server settings have not loaded', () => {
+				const values = { setting1: 'value' };
+				dispatch.setSettings( values );
+
+				// If settings are set on the client, they must be available even
+				// if settings have not been loaded from the server yet.
+				muteConsole( 'error' ); // Ignore the API fetch failure here.
+				expect( select.getSettings() ).toEqual( values );
+			} );
+
+			it( 'dispatches an error if the request fails', async () => {
+				const response = {
+					code: 'internal_server_error',
+					message: 'Internal server error',
+					data: { status: 500 },
+				};
+				fetch
+					.doMockOnceIf(
+						/^\/google-site-kit\/v1\/core\/site\/data\/settings/
+					)
+					.mockResponseOnce(
+						JSON.stringify( response ),
+						{ status: 500 }
+					);
+
+				muteConsole( 'error' );
+				select.getSettings();
+				await subscribeUntil( registry,
+					// TODO: We may want a selector for this, but for now this is fine
+					// because it's internal-only.
+					() => store.getState().isFetchingSettings === false,
+				);
+
+				const settings = select.getSettings();
+
+				expect( fetch ).toHaveBeenCalledTimes( 1 );
+				expect( settings ).toEqual( undefined );
+			} );
+		} );
+
+		describe( 'haveSettingsChanged', () => {
+			it( 'informs whether client-side settings differ from server-side ones', async () => {
+				// Initially false.
+				expect( select.haveSettingsChanged() ).toEqual( false );
+
+				const serverValues = { setting1: 'serverside' };
+				const clientValues = { setting1: 'clientside' };
+
+				fetch
+					.doMockOnceIf(
+						/^\/google-site-kit\/v1\/core\/site\/data\/settings/
+					)
+					.mockResponseOnce(
+						JSON.stringify( serverValues ),
+						{ status: 200 }
+					);
+
+				select.getSettings();
+				await subscribeUntil( registry,
+					() => (
+						select.getSettings() !== undefined
+					),
+				);
+
+				// Still false after fetching settings from server.
+				expect( select.haveSettingsChanged() ).toEqual( false );
+
+				// True after updating settings on client.
+				dispatch.setSettings( clientValues );
+				expect( select.haveSettingsChanged() ).toEqual( true );
+
+				// False after updating settings back to original server value on client.
+				dispatch.setSettings( serverValues );
+				expect( select.haveSettingsChanged() ).toEqual( false );
+			} );
+		} );
+
+		// Tests for "pseudo-selector" getSetting, available via setting-specific "get{SettingSlug}".
+		describe( 'getSetting', () => {
+			it( 'has the correct selector name', () => {
+				expect( Object.keys( storeDefinition.selectors ) ).toEqual(
+					expect.arrayContaining( [ SETTING.selector ] )
```

Comment:
Similarly, I'd find this a bit more informative with the expected contents inline in the test:

```suggestion
					expect.arrayContaining( ['getTestSetting' ] )
```